wagon.py

```diff
@@ -0,0 +1,1015 @@
+
+    We should aspire for the name of the archive to be
+    as compatible as possible with the wheel naming convention
+    described here:
+    https://www.python.org/dev/peps/pep-0491/#file-name-convention,
+    as we're basically providing a "wheel" of our package.
+    """
+    package_name = package_name.replace('-', '_')
+    python_versions = '.'.join(python_versions)
+
+    archive_name_tags = [
+        package_name,
+        package_version,
+        python_versions,
+        'none',
+        platform,
+        'none',
+        'none'
+    ]
+
+    if IS_LINUX and platform != 'any':
+        distro, _, release = _get_os_properties()
+        # TODO: maybe replace `none` with `unknown`?
+        # we found a linux distro but couldn't identify it.
+        archive_name_tags[5] = distro or 'none'
+        archive_name_tags[6] = release or 'none'
+
+    archive_name = '{0}.wgn'.format('-'.join(archive_name_tags))
+    return archive_name
+
+
+def get_source_name_and_version(source):
+    """Retrieves the source package's name and version.
+
+    If the source is a path, the name and version will be retrieved
+    by querying the setup.py file in the path.
+
+    If the source is PACKAGE_NAME==PACKAGE_VERSION, they will be used as
+    the name and version.
+
+    If the source is PACKAGE_NAME, the version will be extracted from
+    the wheel of the latest version.
+    """
+    if os.path.isfile(os.path.join(source, 'setup.py')):
+        package_name, package_version = \
+            _get_name_and_version_from_setup(source)
+    # TODO: maybe we don't want to be that explicit and allow using >=
+    # elif any(symbol in source for symbol in ['==', '>=', '<=']):
+    elif '==' in source:
+        base_name, package_version = source.split('==')
+        package_name = _get_package_info_from_pypi(base_name)['name']
+    else:
+        package_info = _get_package_info_from_pypi(source)
+        package_name = package_info['name']
+        package_version = package_info['version']
+    return package_name, package_version
+
+
+def _create_wagon_archive(format, source_path, archive_path):
+    if format == 'tar.gz':
+        _tar(source_path, archive_path)
+    elif format == 'zip':
+        _zip(source_path, archive_path)
+    else:
+        raise WagonError(
+            'Unsupported archive format to create: {0} '
+            '(Must be one of [zip, tar.gz]).'.format(format))
+
+
+def get_source(source):
+    """Returns a pip-installable source
+
+    If the source is a url to a package's tar file,
+    this will download the source and extract it to a temporary directory.
+
+    If the source is neither a url nor a local path, and is not provided
+    as PACKAGE_NAME==PACKAGE_VERSION, the provided source string
+    will be regarded as the source, which, by default, will assume
+    that the string is a name of a package in PyPI.
+    """
+    def extract_source(source, destination):
+        try:
+            _untar(source, destination)
```

Comment:
I assume the source format can be inferred from the source argument, right? if so there's no need for the try-except blocks below.